application-admintools-default/src/main/java/com/xwiki/admintools/internal/data/ConfigurationDataProvider.java

```diff
@@ -108,44 +108,45 @@ private String getJavaVersion()
     }
 
     /**
-     * Identify the used database for XWiki by verifying the configration files.
+     * Identify the used database for XWiki by accessing the {@link DatabaseMetaData}.
      *
-     * @return the name of the used database or {@code null} in case an error occurred or the used DB is not supported.
+     * @return the name and version of the used database or {@code null} in case an error occurred or the used DB is
+     * not supported.
      */
-    private String identifyDB() throws Exception
+    private String identifyDB()
     {
         String usedDB = null;
         try {
-            ServerIdentifier server = getCurrentServer();
-            String databaseCfgPath = server.getXwikiCfgFolderPath() + "hibernate.cfg.xml";
-            String patternString = "<property name=\"connection.url\">jdbc:(.*?)://";
-            Pattern pattern = Pattern.compile(patternString);
-            File file = new File(databaseCfgPath);
-            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
-
-            while (bufferedReader.ready()) {
-                String line = bufferedReader.readLine();
-                Matcher matcher = pattern.matcher(line);
-                if (matcher.find()) {
-                    String foundDB = matcher.group(1);
-                    usedDB = this.currentServer.getSupportedDBs().getOrDefault(foundDB, null);
-                    break;
-                }
+            DatabaseMetaData metaData = getDBMetaData();
+            if (metaData != null) {
+                usedDB = metaData.getDatabaseProductName() + " - " + metaData.getDatabaseProductVersion();
+            } else {
+                this.logger.warn("Failed to get database metadata.");
             }
-            bufferedReader.close();
-            if (usedDB == null) {
-                this.logger.warn("Failed to find database. Used database may not be supported!");
-            }
-        } catch (NullPointerException e) {
-            throw new Exception("Failed to identify used Database.", e);
+        } catch (SQLException e) {
+            this.logger.warn("Failed to compute database name and version. Root cause is: [{}]",
+                ExceptionUtils.getRootCauseMessage(e));
         } catch (Exception exception) {
-            this.logger.warn("Failed to open database configuration file. Root cause is: [{}]",
+            this.logger.warn("Failed to get database metadata. Root cause is: [{}]",
                 ExceptionUtils.getRootCauseMessage(exception));
         }
-
         return usedDB;
     }
 
+    private DatabaseMetaData getDBMetaData()
```

Comment:
Why can't you use the DatabasePingDataProvider, as you did for servlet?

Reply:
Initially, I wanted to add custom messages, but I took a look again over DatabasePingDataProvider and saw that it's almost similar. Adapted implementation  to use DatabasePingDataProvider.